arcgis-ios-sdk-samples/Maps/Display layer view state/LayerStatusViewController.swift

```diff
@@ -99,44 +81,39 @@ class LayerStatusViewController: UIViewController, UITableViewDataSource, UITabl
         if status.contains(.error) {
             statuses.append("Error")
         }
+        if status.contains(.warning) {
+            statuses.append("Warning")
+        }
         if !statuses.isEmpty {
             return statuses.joined(separator: ", ")
         } else {
             return "Unknown"
         }
     }
     
-    // MARK: - Table view data source
+    // MARK: Actions
     
-    func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+    @IBAction func layerVisibilitySwitchValueChanged(_ sender: UISwitch) {
+        featureLayer.isVisible = sender.isOn
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.map?.operationalLayers.count ?? 0
-    }
+    // MARK: UIViewController
     
-    // MARK: - Table view delegates
-    
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "LayerStatusCell", for: indexPath)
-        cell.backgroundColor = .clear
-        let layer = self.map.operationalLayers[indexPath.row] as! AGSLayer
-        
-        //if the layer is loaded then show the name
-        //else use a template
-        if layer.loadStatus == .loaded {
-            cell.textLabel?.text = layer.name
-        } else {
-            cell.textLabel?.text = "Layer \(indexPath.row)"
-        }
-        
-        if let layerViewState = mapView.layerViewState(for: layer) {
-            cell.detailTextLabel?.text = viewStatusString(layerViewState.status)
-        } else {
-            cell.detailTextLabel?.text = "Unknown"
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames  = ["LayerStatusViewController"]
+        // Avoid the overlap between the status label and the map content.
+        mapView.contentInset.top = 2 * statusLabel.font.lineHeight
+        // Load the feature layer.
+        featureLayer.load { [weak self] (error) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                self.mapView.map?.operationalLayers.add(self.featureLayer)
+                self.mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -11e6, y: 45e5, spatialReference: .webMercator()), scale: 2e7))
```

Comment:
Since this is being set after the feature layer loads, the map initially displays at one viewpoint and then jumps to this one when the feature layer finishes loading. It would be better to either set this in the `mapView` property observer or set the map's `initialViewpoint` property to the desired viewpoint.